src/sinks/uds.rs

```diff
@@ -0,0 +1,197 @@
+/// a UDS socket and then the metric is written to the buffer. The buffer is
+/// also flushed when this sink is destroyed.
+///
+/// The default size of the buffer is 512 bytes. This is to be consistent with
+/// the default for the BufferedUdpMetricSink. The buffer size can be customized
+/// using the `with_capacity` method to create the sink if desired.
+///
+/// If a metric larger than the buffer is emitted, it will be written
+/// directly to the underlying UDS socket, bypassing the buffer.
+#[derive(Debug)]
+pub struct BufferedUdsMetricSink {
+    buffer: Mutex<MultiLineWriter<UnixStream>>,
+}
+
+impl BufferedUdsMetricSink {
+    /// Construct a new `BufferedUdsMetricSink` instance with a default
+    /// buffer size of 512 bytes.
+    ///
+    /// The socket should already be bound to a local address with any desired
+    /// configuration applied (blocking vs non-blocking, timeouts, etc.).
+    ///
+    /// Writes to this sink are automatically suffixed with a Unix newline
+    /// ('\n') by the sink and stored in a 512 byte buffer until the buffer
+    /// is full or this sink is destroyed, at which point the buffer will be
+    /// flushed.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use std::os::unix::net::UnixStream;
+    /// use cadence::BufferedUdsMetricSink;
+    ///
+    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
+    /// let sink = BufferedUdsMetricSink::from(socket);
+    /// ```
+    pub fn from(socket: UnixStream) -> BufferedUdsMetricSink {
+        Self::with_capacity(socket, DEFAULT_BUFFER_SIZE)
+    }
+
+    /// Construct a new `BufferedUdsMetricSink` instance with a custom
+    /// buffer size.
+    ///
+    /// The socket should already be bound to a local address with any desired
+    /// configuration applied (blocking vs non-blocking, timeouts, etc.).
+    ///
+    /// Writes to this sink are automatically suffixed  with a Unix newline
+    /// ('\n') by the sink and stored in a buffer until the buffer is full
+    /// or this sink is destroyed, at which point the buffer will be flushed.
+    ///
+    /// For guidance on sizing your buffer see the
+    /// [Statsd docs](https://github.com/etsy/statsd/blob/master/docs/metric_types.md#multi-metric-packets).
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use std::os::unix::net::UnixStream;
+    /// use cadence::BufferedUdsMetricSink;
+    ///
+    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
+    /// let sink = BufferedUdsMetricSink::with_capacity(socket, 1432);
+    /// ```
+    pub fn with_capacity(socket: UnixStream, cap: usize) -> BufferedUdsMetricSink {
+        BufferedUdsMetricSink {
+            buffer: Mutex::new(MultiLineWriter::new(cap, socket)),
+        }
+    }
+}
+
+impl MetricSink for BufferedUdsMetricSink {
+    fn emit(&self, metric: &str) -> io::Result<usize> {
+        let mut writer = self.buffer.lock().unwrap();
+        writer.write(metric.as_bytes())
```

Comment:
Per #87 I realized that `MultiLineWriter` will do something wrong-ish when given data to write that is larger than its buffer (it writes the data and trailing newline in two `.write()` calls which is wrong for a UDP socket). I decided in #87 that in this case, the trailing newline will be omitted since it doesn't make sense for UDP.

Can you confirm that this proposed change won't cause issues with the `UnixStream` in use here?